app/routes.py

```diff
@@ -1,2 +1,223 @@
+    task = Task.query.get(task_id)
+    
+    if task == None:
+        return make_response("",404)
+
+    task.completed_at = None
+    db.session.commit()
+    
+    return make_response({"task" : task.to_json()}), 200
+
+
+
+# wave 5 
+
+@goals_bp.route("", methods=["GET", "POST"])
+def all_goals():    
+    if request.method == "POST":
+        request_body = request.get_json()
+    # Create a Goal: Invalid title or With Missing Data
+        if "title" not in request_body:
+            return make_response ({"details": "Invalid data"}),400
+        else:
+            new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal)
+        db.session.commit()
+        return make_response({"goal": new_goal.to_json()}), 201
+    elif request.method == "GET":
+        goals = Goal.query.all()
+        
+        goals_response =[]
+        for goal in goals:
+            goals_response.append(goal.to_json())
+        
+        return jsonify(goals_response)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def one_goal_only(goal_id):    
+    goal = Goal.query.get(goal_id)
+    
+    # GET/PUT/DELETE request to /goals/1 when there are no matching goals and get this response
+    if goal == None:
+        return make_response("",404)
+
+    if request.method == "GET":
+        return make_response({"goal" : goal.to_json()}), 200
+    elif request.method == "PUT":
+        # converting postman json to a dict table
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+
+        db.session.commit()
+        return make_response({"goal" : goal.to_json()}), 200
+    
+    elif request.method == "DELETE":
+# DELETE requests do not generally include a request body, so 
+# no additional planning around the request body is needed        
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response ({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'})
+
+# wave 6 
+@goals_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])
+def one_goal_many_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal == None:
+        return make_response("",404)
+    # the request body will include task_ids and link them to the goal_id
+    # listed on the path 
+    if request.method == "POST":
+        request_body = request.get_json()
+        for one_task_id in request_body["task_ids"]:
+            task=Task.query.get(one_task_id)
+            task.goal_id = goal.goal_id
+        return make_response({
+            "id":goal.goal_id,
+            "task_ids":request_body["task_ids"]}), 200
+    elif request.method == "GET":
+        tasks = Task.query.filter_by(goal_id=goal_id)
+        task_list =[]
+        for task in tasks:
+            task_list.append(task.goal_json())
+        return make_response({
+            "id":goal.goal_id,
+            "title":goal.title,
+            "tasks":task_list
+        })
```

Comment:
Consider making a goal instance method to build this response as you have for other responses. Even if a method is only used once, it helps add readability, changeability, and consistency to encapsulate that functionality in an instance method.